### PR TITLE
Add VB.Net Sample Project

### PR DESCRIPTION
**Pull request by jrmoreno1**
Sample VB project based upon the C# project. Doesn't include all of the examples that the C# project does, but with a working example, those should be simple extensions to what has already been done.

### CODE REVIEW

File: SampleWebApplicationVB/Default.aspx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿<%@ Page Title="Home Page" Language="VB"
```
Comment (Jehoel) [inline SampleWebApplicationVB/Default.aspx:1]: Please use consistent formatting.

File: SampleWebApplicationVB/Web.Release.config [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿<?xml version="1.0"?>
```
Comment (Jehoel) [inline SampleWebApplicationVB/Web.Release.config:1]: As this runs locally, we can remove the `Web.Release.config` and `Web.Debug.config` transformations.

File: AspNetDependencyInjection.sln [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.28307.757
+# Visual Studio Version 16
```
[the remark this answers is not in the document]
Comment (Jehoel) [reply AspNetDependencyInjection.sln:3]: Also, please add an automated build script for the project (ensure it runs both `MSBuild.exe` to build the `.vbproj` and `aspnet_compiler` to precompile the `.aspx` files too)

File: SampleWebApplicationVB/AppCode/DevHttpClientHandler.vb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+Public Class DevHttpClientHandler
+    Inherits HttpClientHandler
+
+
+    Public Sub New()
+
+    End Sub
+
+    Protected Overrides Function SendAsync(request As System.Net.Http.HttpRequestMessage, cancellationToken As Threading.CancellationToken) As Task(Of HttpResponseMessage)
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/AppCode/DevHttpClientHandler.vb:13]: Done locally, will push soon.

File: SampleWebApplicationVB/App_Start/SampleApplicationStart.vb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    Private Sub ConfigureServices(services As IServiceCollection)
+        ' TODO: Add any dependencies needed here
+        services = services.AddDefaultHttpContextAccessor()
+
+        Dim devMode As Boolean = True
+        Dim apiBase As String = "https://github.com/Jehoel/AspNetDependencyInjection"
+
+        services.AddHttpClient("exampleFactory", Sub(c As HttpClient)
+                                                     c.BaseAddress = New Uri(apiBase)
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/App_Start/SampleApplicationStart.vb:36]: Yes, I just tested it, typing everything in by hand, and I get that exact indenting when I hit return.  I can override it to however you like if you want, but I would have to use a different editor or turn off Visual Studio's VB pretty printer as it reformats the code at any change.

File: SampleWebApplicationVB/Default.aspx.vb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+Imports System.Web.UI
+
+
+''' <summary>
+'''		Sample page demonstrating injection at the page level.
+''' <'summary>
+Partial Public Class DefaultPage
+	Inherits Page
+
+	Private HttpClientFactory As IHttpClientFactory
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/Default.aspx.vb:12]: Done locally.

File: SampleWebApplicationVB/Default.aspx.vb [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+	Public Sub New(httpClientFactory As IHttpClientFactory)
+		Me.HttpClientFactory = httpClientFactory
+	End Sub
+
+	Protected Overrides Sub OnInit(e As EventArgs)
+		If (Not IsPostBack) Then
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/Default.aspx.vb:19]: Done locally.

File: SampleWebApplicationVB/Global.asax.vb [hunk 8]
```diff
@@ -0,0 +1,43 @@
+﻿Imports System
+Imports System.Collections.Generic
+Imports System.Linq
+Imports System.Web
+Imports System.Web.Security
+Imports System.Web.SessionState
+
+
+Public Class SampleWebApplicationVBHttpApplication
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/Global.asax.vb:9]: Done locally

File: SampleWebApplicationVB/SampleWebApplicationVB.vbproj [hunk 9]
```diff
@@ -0,0 +1,247 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/SampleWebApplicationVB.vbproj:1]: I have consolidated the itemgroups and removed unnecessary references (kept all references that the c# project had).

File: SampleWebApplicationVB/Site.Master [hunk 10]
```diff
@@ -0,0 +1,38 @@
+﻿<%@ Master Language="VB" AutoEventWireup="true" CodeBehind="Site.master.vb" Inherits="SampleWebApplicationVB.SiteMaster" %>
+
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
+<head runat="server">
+	<title>WebForms</title>
+
+	<link href="~/Styles/Site.css" rel="stylesheet" type="text/css" />
+
+	<asp:ContentPlaceHolder ID="HeadContent" runat="server"></asp:ContentPlaceHolder>
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/Site.Master]: done

File: SampleWebApplicationVB/Web.config [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+		</roleManager>
+		<pages controlRenderingCompatibilityVersion="4.0" />
+	</system.web>
+	
+	<system.webServer>
+		<modules runAllManagedModulesForAllRequests="true" />
+	</system.webServer>
+	
+	<runtime>
+				<assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/Web.config]: done.

File: SampleWebApplicationVB/packages.config [hunk 12]
```diff
@@ -0,0 +1,28 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Jehoel.AspNetDependencyInjection" version="4.0.0" targetFramework="net472" />
```
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply SampleWebApplicationVB/packages.config:3]: Removed unused packages. There are now 5 packages that show they could be updated in the VB project, 3 of them are shared with the C# project.  Don't want to get those out of sync.  System.Runtime goes up a point 4.3.0 to 4.3.1, I used an older version of Microsoft.Extensions.Http so as to not pull in extra dependencies.  I can update these if you like.

File: AspNetDependencyInjection.sln (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrmoreno1) [reply AspNetDependencyInjection.sln]: I'm not sure what you mean by adding a build script.  Do you have an example in the solution I could use as a template?

File: SampleWebApplicationVB/Web.Release.config (hunk 2 above)
Comment (jrmoreno1) [reply SampleWebApplicationVB/Web.Release.config]: Done

File: SampleWebApplicationVB/packages.config (hunk 12 above)
[the remark this answers is not in the document]
Comment (Jehoel) [reply SampleWebApplicationVB/packages.config:3]: Keeping them in-sync with the current dependencies is preferable, actually - thank you for pointing that out. I'll update all referenced package versions when I make the next release.

File: AspNetDependencyInjection.sln (hunk 3 above)
[the remark this answers is not in the document]
Comment (Jehoel) [reply AspNetDependencyInjection.sln]: The quick-and-dirty explanation is to open the VS "Developer Command Prompt" and `cd` into the solution root and run `msbuild SampleWebApplicationVB\SampleWebApplicationVB.vbproj` and ensure it works. If that doesn't work then create a `Build.cmd` file (like in `$\nuget\Build.cmd`) that performs the necessary preparatory work before running `msbuild SampleWebApplicationVB\SampleWebApplicationVB.vbproj`.

[the remark this answers is not in the document]
Comment (jrmoreno1) [reply AspNetDependencyInjection.sln]: I ran: msbuild SampleWebApplicationVB\SampleWebApplicationVB.vbproj and final output was:
Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:02.53

**Comment by jrmoreno1** [comment]
I believe the only outstanding change would be the updating of Microsoft.Extensions.Http to a more current version, I think that ought to wait until you're ready to update all Microsoft.Extensions.Http, and it seemed from your comment that you agreed.  So I believe that is it.

**Pull request merged**